builtin/plugins/dposv3/dpos.go

```diff
@@ -1495,6 +1567,16 @@ func rewardAndSlash(ctx contract.Context, cachedDelegations *CachedDposStorage,
 			delegatorRewards[validatorKey] = common.BigZero()
 			formerValidatorTotals[validatorKey] = *common.BigZero()
 		} else {
+
+			// If a validator is jailed, don't calculate and distribute rewards
+			if ctx.FeatureEnabled(loomchain.DPOSVersion3_3, false) {
+				if statistic.Jailed {
+					delegatorRewards[validatorKey] = common.BigZero()
+					formerValidatorTotals[validatorKey] = *common.BigZero()
+					continue
```

Comment:
Note that this `continue` makes you skip over the slashing call https://github.com/loomnetwork/loomchain/pull/1250/files#diff-1f3fbb833197fcdf09e638ef5fccb479R1656. Is that desired?

Reply:
We should be slashing the validator before jailing them (or roughly at the same point) so that this continue should fine. Additional slashing should not happen post-jailing, only after a jailed validator unjails themselves and commits another fault.